src/blenderbim/blenderbim/bim/module/pset/ui.py

```diff
@@ -375,3 +375,92 @@ def draw(self, context):
 
         for pset in WorkSchedulePsetsData.data["psets"]:
             draw_psetqto_ui(context, pset["id"], pset, props, self.layout, "WorkSchedule")
+
+
+class BIM_PT_BulkPropertyEditor(Panel):
+    bl_label = "IFC Bulk Property Editor"
+    bl_space_type = 'PROPERTIES'
+    bl_region_type = 'WINDOW'
+    bl_context = "object"
+
+    def draw(self, context):
+        pass
+    
+
+class BIM_PT_RenameParameters(Panel):
+    bl_label = "Rename all building elements"
+    bl_space_type = 'PROPERTIES'
+    bl_region_type = 'WINDOW'
+    bl_context = "object"
+    bl_parent_id = "BIM_PT_BulkPropertyEditor"
+    bl_options = {"DEFAULT_CLOSED"}
+    bl_order = 0
+    
+    def draw(self, context):
+        layout = self.layout
+        props = context.scene.PropertiesToRename
+
+        if props:          
+            for index, property in enumerate(props):
+                row = layout.row()
+                row.prop(property, "pset_name", text="")
+                row.prop(property, "existing_property_name", text="")
+                row.prop(property, "new_property_name", text="")
+                op = row.operator("bim.remove_property_to_edit",icon="PANEL_CLOSE", text="")
+                op.index = index
+                op.type = "properties_to_map"
+                
+        row = layout.row()
+        row.label()
+        op = row.operator("bim.add_property_to_edit", icon="ADD",text="")
+        op.type = "PropertiesToRename"
+        
+        if props:  
+            row = layout.row()
+            clear = row.operator("bim.clear_list")
+            clear.type = "PropertiesToRename"
+            row.operator("bim.rename_parameters")
+
+
+class BIM_PT_AddPropertiesOrEditValues(Panel):
+    """Creates a Panel in the Object properties window"""
+    bl_label = "Add/Edit Custom Properties and Values (to selected objects)"
+    bl_space_type = 'PROPERTIES'
+    bl_region_type = 'WINDOW'
+    bl_context = "object"
+    bl_parent_id = "BIM_PT_BulkPropertyEditor"
+    bl_options = {"DEFAULT_CLOSED"}
+    bl_order = 1
+
+    def draw(self, context):
+        layout = self.layout
+        props = context.scene.PropertiesToAddOrEdit 
+        
+        if props:
+            for index, property in enumerate(props):
+                    row = layout.row()
+                    row.prop(property, "pset_name", text="")
+                    row.prop(property, "property_name", text="")
+                    if property.value_type == "String":
```

Comment:
Similar to previous comment : Maybe worthwhile to explore the [Attribute](https://github.com/IfcOpenShell/IfcOpenShell/blob/3accc1dbd6510d176455e8f163fdbfe157762bdc/src/blenderbim/blenderbim/bim/prop.py#L160-L213) property group : you can then just do `row.prop(property, property.get_value_name(), text="")` and keeping the `value_type` enum to dynamically change it afterwards with an `update` method to change the attribute `data_type`. https://docs.blender.org/api/current/bpy.props.html#update-example

Reply:
Yeah, sounds good.